### PR TITLE
Added file system caching to import resolver for performance reasons.

### PR DESCRIPTION
**Pull request by erictraut**

### CODE REVIEW

**Comment by erictraut** [comment]
I confirmed this gives us the same perf benefit. I think this is cleaner than the previous one. Definitely simpler and more contained.

File: packages/pyright-internal/src/analyzer/importResolver.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@
  * runtime rules of Python.
  */
 
+import * as fs from 'fs';
```
Comment (jakebailey) [inline packages/pyright-internal/src/analyzer/importResolver.ts:11]: This is a nit, but just to keep it clear that this file doesn't actually access the filesystem directly, can you do a `import type { Dirent } from 'fs'` or similar?

Comment (jakebailey) [reply packages/pyright-internal/src/analyzer/importResolver.ts:11]: `fs` is also a variable used in a few places, so this may get shadowed.

**Review by heejaechang — APPROVED** [review]

File: packages/pyright-internal/src/pyrightFileSystem.ts [hunk 2]
```diff
@@ -70,7 +74,11 @@ export class PyrightFileSystem implements FileSystem {
     }
 
     readdirSync(path: string): string[] {
-        const entries = this._realFS.readdirSync(path);
+        const entries = this._realFS.readdirSync(path).filter((item) => {
```
Comment (heejaechang) [inline packages/pyright-internal/src/pyrightFileSystem.ts:77]: thank you!

**Pull request merged**